api/v1/discoveryconfig_types.go

```diff
@@ -38,6 +38,10 @@ type DiscoveryConfigSpec struct {
 	// +optional
 	ProviderConnections []string `json:"providerConnections,omitempty"`
 
+	// Credential is the secret containing credentials to connect to the OCM api on behalf of a user
+	// +optional
```

Comment:
Perhaps this should be mandatory? 
Is there a point to a discoveryconfig w/o a credential?

Reply:
Removed optional flag